mkdocs.yml

```diff
@@ -5,11 +5,25 @@ repo_name: alandefreitas/small
 
 site_description: "Small: C++ Small Containers"
 
+nav:
+  - Home: index.md
+  - Quickstart: quickstart.md
+  - Vectors: vectors.md
+  - Strings: strings.md
+  - Sets and Maps: sets_and_maps.md
+  - Reference: # Reference is organized by module
```

Comment:
I'm sorry for the delayed response. For sure we need to figure this out. So the `small::detail` shouldn't appear in the documentation, right?

Reply:
Exactly. 👍️

The `detail` namespace is always private.

Reply:
> I did some changes and the namespace detail is not appearing anymore in the docs. 

Great. `detail` is only in the index now, when talking about sets and maps. But we do need a better solution for sets and maps anyway. 

- The best solution we could have for now (creating another issue) is to move the `detail`s that need to get exposed to another namespace, like `exposition_only` or something, so that Doxygen can render these for us. In the doxybook templates, we could include a big warning about this being exposition only whenever we render a class in this namespace. At least we would have something in the reference, because sets and maps are completely missing from the reference now.
- Something that would also help, regardless, is to define the doxygen modules (creating another issue). The subsections of the reference should refer to modules and not classes because there are lots of free functions in the vector/string API and these are not rendered at the moment (to_string, stol, ...).
- The best long-term solution, would be to completely refactor that when improving these associative containers, which needs to happen at some point.

> I didn't understand exactly the problem with is_relocatable. 

`is_relocatable` is now at a different position, so the previous comment doesn't completely apply anymore. It's now only listed in "Classes" and "Index". One problem now is that only the `is_relocatable< small::basic_string< CharT, N, Traits, WCharT, Allocator, CP_HINT_STEP, SizeType > >` specialization is appearing in the docs. I don't know why the main class is not in the docs, but I would say we should remove `is_relocatable` from the docs for now in that case. Because the list of classes as `basic_string`, `is_relocatable< small::basic_string< CharT, N, Traits, WCharT, Allocator, CP_HINT_STEP, SizeType > >`, and `vector` is not looking very good. We could find out why this is not rendering later and include it in this vector module.

> I added two Sections in the TOC with vector and basic_string

Yes. That improved things a lot. 

- It's natural to call the string class/module only string instead of basic_string. basic_string is almost like an implementation detail. It's meant to be used by implementers to generalize string types and used rarely by very advanced users. For this reason, cppreference simply calls this the "string" library, which is what we can use in the TOC.
- Since small::string depends on small::vector, it's natural that vector comes first in the TOC.

Also, I might be doing something wrong, but the very first subsection in the reference 

```yml
- Small: reference/Modules/index.md        # futures module
```

is still completely empty: `http://127.0.0.1:8000/small/reference/Modules/`.

There's no point in a modules section since we have no modules (for now), let alone the very first section. There are also some references to "futures" in `mkdocs.yml`.

> to open an exception for detail::associative_vector. 

Including a detail in the public API, especially as the only class directly in the TOC is not a good idea because users are not supposed to use anything from the `detail` namespace. This is part of the contract. We could do the other `exposition_only` solution as a first step. We could do it in this PR or open another issue. I leave it up to you.